samples/nrf9160/http_application_update/src/main.c

```diff
@@ -97,10 +117,78 @@ static void app_dfu_transfer_start(struct k_work *unused)
 					     DT_GPIO_PIN(DT_ALIAS(sw0), gpios),
 					     GPIO_INT_EDGE_TO_ACTIVE);
 
+		printk("fota_download_start() failed, err %d\n", retval);
+		return;
+	}
+
+}
+
+static void apply_fmfu(void)
+{
+	int err;
+
+	err = fmfu_fdev_load(0);
+	if (err != 0) {
+		printk("fmfu_fdev_load failed: %d\n", err);
+		return;
+	}
+
+	printk("Modem firmware update completed, please restart the device");
+
+}
+
+static void fmfu_transfer_start(struct k_work *unused)
+{
+	const struct device *flash_dev;
+	const char *file = CONFIG_DOWNLOAD_MODEM_0_FILE;
+	int err;
+	char *apn = NULL;
+
+	printk("Started full modem firmware update\n");
+
+	flash_dev = device_get_binding(EXT_FLASH_DEVICE);
+
+	/* Pass 0 as size since that tells the stream_flash module
+	 * that we want to use all the available flash in the device
+	 */
+	err = dfu_target_modem_full_cfg(fota_buf, sizeof(fota_buf), flash_dev,
+					0, 0);
+	if (err != 0) {
+		printk("dfu_target_modem_full_cfg failed: %d\n", err);
+		return;
+	}
+
+	/* We can re-use the fota_buf as fmfu_fdev is only executed once
+	 * the whole update has been downloaded.
+	 */
+	err = fmfu_fdev_init(fota_buf, sizeof(fota_buf), flash_dev);
```

Comment:
Yes. I'm on the fence about this already, since the buffer might be easier available (as in scope) in a different place from where you want to call `fmfu_fdev_load` so my thought was that this made it easier for the user to assign the buffer from wherever it is available.

Reply:
Will pass it to _load